src/api/types.ts

```diff
@@ -0,0 +1,116 @@
+import { Stream } from '../util/grpc'
+
+export type JSONString = string
+export type hash = string
+
+export type Event = {
+  hash?: hash
+  instanceHash: string
+  key: string
+  data: string
+}
+
+export enum ExecutionStatus {
+  UNKNOWN = 0,
+  CREATED = 1,
+  IN_PROGRESS = 2,
+  COMPLETED = 3,
+  FAILED = 4,
+}
+export type Execution = {
+  hash?: hash
+  parentHash: hash
+  eventID: string
+  status: ExecutionStatus
+  instanceHash: string
+  taskKey: string
+  inputs: JSONString
+  outputs?: JSONString
+  error?: string
+  tags?: string[]
+}
+
+export type Instance = {
+  hash?: hash
+  serviceHash: hash
+}
+
+export type Service = {
+  hash?: hash
+  sid: string
+  name?: string
+  description?: string
+  configuration?: any
+  tasks?: any[]
+  events?: any[]
+  dependencies?: any[]
+  repository?: string
+  source: string
+}
+
+export type EventCreateInputs = { event: Event }
+export type EventCreateOutputs = Promise<{ hash: hash }>
+
+export type EventStreamInputs = { filter?: { hash?: string, instanceHash?: string, key?: string } }
+export type EventStreamOutputs = Stream<Event>
```

Comment:
Is it really good to create new type in this case?
Same for all `Promise<AlreadyCreatedType>`?
I feel it more logic to move this in the return type of functions.

Reply:
The good point here is that if we want to change and not have a stream of the event because we want for example to add some metadata on this event then we can just change this type and we don't have to change it everywhere (including the applications)